database/migrations/2022_04_18_174417_fix_live_photo_short_path.php

```diff
@@ -0,0 +1,68 @@
+<?php
+
+use Doctrine\DBAL\Exception as DBALException;
+use Illuminate\Database\Migrations\Migration;
+use Illuminate\Support\Facades\DB;
+use Illuminate\Support\Facades\Schema;
+
+class FixLivePhotoShortPath extends Migration
+{
+	private string $driverName;
+
+	/**
+	 * @throws DBALException
+	 */
+	public function __construct()
+	{
+		$connection = Schema::connection(null)->getConnection();
+		$this->driverName = $connection->getDriverName();
+	}
+
+	/**
+	 * Run the migrations.
+	 *
+	 * @return void
+	 *
+	 * @throws RuntimeException
+	 */
+	public function up()
+	{
+		// MySQL misuses the ANSI SQL concatenation operator `||` for
+		// a logical OR and provides the proprietary `CONCAT` statement
+		// instead.
+		$sqlConcatLivePhotoPath = match ($this->driverName) {
+			'mysql' => DB::raw('CONCAT(\'big/\', live_photo_short_path)'),
+			'pgsql', 'sqlite' => DB::raw('\'big/\' || live_photo_short_path'),
+			default => throw new \RuntimeException('Unknown DBMS')
+		};
+
+		DB::table('photos')
+			->whereNotNull('live_photo_short_path')
+			->where('live_photo_short_path', 'not like', '%/%')
+			->update(['live_photo_short_path' => $sqlConcatLivePhotoPath]);
+	}
+
+	/**
+	 * Reverse the migrations.
+	 *
+	 * @return void
+	 *
+	 * @throws RuntimeException
+	 */
+	public function down()
+	{
+		// In contrast to all other programming languages, the first character
+		// of a string has index 1 (not 0) in SQL.
```

Comment:
You've never programmed in Pascal, have you? :wink:

Reply:
I did and forgot. 😂 I had Pascal at middle school.